### PR TITLE
clevis: add new package

### PR DESCRIPTION
**Pull request by patrykk**
Automated Encryption Framework supports tang key server and LUKS volumes.

Maintainer: @patrykk
Compile tested: (miediatek filogic ARMv8, Asus TUF AX6000; Asus TUF AX4200, OpenWrt main branch 22.09.2023)
Run tested: (miediatek filogic ARMv8, Asus TUF AX6000; Asus TUF AX4200 , OpenWrt main branch 22.09.2023, tests done: run scripts; luks devices mounted with network tang)

Description:

We can automate unlock the LUKS block devices with network key server tang.

### CODE REVIEW

File: utils/clevis/patches/001-no-old-cryptsetup.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+--- a/src/luks/tests/meson.build	2023-09-21 18:25:28.574733087 +0200
++++ b/src/luks/tests/meson.build	2023-09-21 18:25:48.804764217 +0200
```
Comment (BKPepe) [inline utils/clevis/patches/001-no-old-cryptsetup.patch:2]: Why do we need to have this patch? Who is the author? Is this backport from upstream repository?

**Review by BKPepe — CHANGES_REQUESTED** [review]

File: utils/clevis/Makefile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=clevis
+PKG_VERSION:=19
+PKG_RELEASE:=1
+
+PKG_SOURCE:=v$(PKG_VERSION).tar.gz
```
[the remark this answers is not in the document]
Comment (patrykk) [reply utils/clevis/Makefile:14]: Hi. Do you have any advice howto do that when source package name contain only version ?

File: utils/clevis/Makefile [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#
+# Copyright (C) 2023 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
```
[the remark this answers is not in the document]
Comment (patrykk) [reply utils/clevis/Makefile:6]: Ok. I'll remove it.

File: utils/clevis/patches/001-no-old-cryptsetup.patch (hunk 1 above)
Comment (patrykk) [reply utils/clevis/patches/001-no-old-cryptsetup.patch:2]: I have had error like below: 
 "ERROR: Entry OLD_CRYPTSETUP not in configuration data."

File: utils/clevis/Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (BKPepe) [reply utils/clevis/Makefile:14]: No, it's not. It downloads ``clevis-19.tar.gz``.

Look at other packages, which we do have it in this repository how it needs to be done or even you can use documentation.

File: utils/clevis/patches/001-no-old-cryptsetup.patch (hunk 1 above)
Comment (BKPepe) [reply utils/clevis/patches/001-no-old-cryptsetup.patch:2]: Did you discussed it with upstream?

Comment (patrykk) [reply utils/clevis/patches/001-no-old-cryptsetup.patch:2]: No. I'll investigate the problem and will try to disable all tests during build process.

Comment (BKPepe) [reply utils/clevis/patches/001-no-old-cryptsetup.patch:2]: Anything new?

**Pull request closed without merging**